samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -242,63 +279,18 @@ int main() {
-#ifdef PROFILE_SYCL
-      auto i = n + m * num_host_tile_n;
-      event.wait();
-      auto end = std::chrono::system_clock::now();
-      std::chrono::duration<double, std::milli>
-          current_application_execution_time = end - start;
-      total_application_execution_time +=
-          current_application_execution_time.count();
-      auto submit_time = event.get_profiling_info<
-          cl::sycl::info::event_profiling::command_submit>();
-      auto start_time = event.get_profiling_info<
-          cl::sycl::info::event_profiling::command_start>();
-      auto end_time = event.get_profiling_info<
-          cl::sycl::info::event_profiling::command_end>();
-      auto current_submission_time = (start_time - submit_time) / 1000000.0f;
-      auto current_execution_time = (end_time - start_time) / 1000000.0f;
-
-      total_execution_time += current_execution_time;
-      total_submission_time += current_submission_time;
-#ifdef PER_EVENT_PROFILING
-      std::cout << "event, " << i << " , current_kernel_submission_time, "
-                << current_submission_time
-                << " , current_kernel_execution_time, "
-                << current_execution_time
-                << ", current_application_execution_time, "
-                << current_application_execution_time.count() << "\n";
-#endif
-#endif
       host_offset_n += mat_size.n;
     }
     host_offset_m += mat_size.m;
   }
-// this part is used to profile the total execution time
 #ifdef PROFILE_SYCL
-  avarage_submission_time =
-      total_submission_time / float(num_host_tile_n * num_host_tile_m);
-  avarage_execution_time =
-      total_execution_time / float(num_host_tile_n * num_host_tile_m);
-  avarage_application_execution_time = total_application_execution_time /
-                                       float(num_host_tile_n * num_host_tile_m);
-  std::cout << " local_size_0, " << ocl_config_t::local_size_m
-            << " , local_size_1, " << ocl_config_t::local_size_n
-            << " , total_kernel_submission_time, " << total_submission_time
-            << " , total_kernel_execution_time, " << total_execution_time
-            << " , avarage_kernel_submission_time, " << avarage_submission_time
-            << " , avarage_kernel_execution_time, " << avarage_execution_time
-            << " , total_application_execution_time, "
-            << total_application_execution_time
-            << " , avarage_application_execution_time, "
-            << avarage_application_execution_time << "\n";
+  profiler(events, starts);
 #endif
   // check the correctness of the result
   auto out_data = out_buff.get_access<read_t>();
   for (index_t m = 0; m < total_buffer.m; m++) {
-    for (index_t n = 0; n < total_buffer.m; n++) {
-      if (!(std::abs(index_t(out_data[m][n] - (input_data * filter_data))) <
+    for (index_t n = 0; n < total_buffer.n; n++) {
+      if (!(std::fabs(index_t(out_data[m][n] - (input_data * filter_data))) <
```

Comment:
You still have the cast to index_t around the expression, this will still likely truncate to zero and lose the erroneous values.